app/core/models/ocelot/stanzas/ChoiceStanza.scala

```diff
@@ -113,7 +113,22 @@ case class ContainsTest(left: String, right: String) extends ChoiceTest {
   def contains(l: BigDecimal, r: BigDecimal): Boolean = l.toString.toLowerCase().contains(r.toString.toLowerCase())
   def contains(l: String, r: String): Boolean = l.toLowerCase().contains(r.toLowerCase())
   def contains(l: LocalDate, r: LocalDate): Boolean = l.toString.toLowerCase().contains(r.toString.toLowerCase())
-  def eval(labels: Labels): Boolean = op(contains(_, _), contains(_, _), contains(_, _), labels)
+  def eval(labels: Labels): Boolean =
+    labelReference(left).fold(op(contains(_, _), contains(_, _), contains(_, _), labels)){lbl =>
+      labels.valueAsList(lbl).fold(op(contains(_, _), contains(_, _), contains(_, _), labels)){l =>
+        l.exists{el =>
+          val y = value(right, labels)
+          (asDate(el), asDate(y)) match {
+            case (Some(ld1), Some(ld2)) => contains(ld1, ld2)
+            case _ =>
+              (asDecimal(el), asDecimal(y)) match {
+                case (Some(bd1), Some(bd2)) => contains(bd1, bd2)
+                case _ => contains(el.toString, y.toString)
```

Comment:
It might be better to use mkString(", ")

Reply:
el is an element of the list

Reply:
Sorry, I mistakenly thought exists was working on an Option rather than the list. So it looks ok. However, it might be worth moving the definition of y outside the exists loop.

Reply:
done